## List ALL available models (there are 1000+)
Total number of models in timm:


In [10]:
import warnings
warnings.filterwarnings("ignore", category=UserWarning, module="pydantic._internal._generate_schema")

import timm
all_models = timm.list_models()
print(f"Total: {len(all_models)} models")

Total: 1200 models


##  List only pretrained models


In [11]:
print("\n Pretrained models:")
pretrained_models = timm.list_models(pretrained=True)
print(f"   Total pretrained: {len(pretrained_models)} models")


 Pretrained models:
   Total pretrained: 1599 models


## Search for specific model types


In [12]:
print("\n Search for specific models:")
print("\n   a) Vision Transformers (ViT):")
vit_models = timm.list_models('*vit*', pretrained=True)
print(f"      Found {len(vit_models)} ViT models")
for i, model in enumerate(vit_models[:5]):
    print(f"      - {model}")
if len(vit_models) > 5:
    print(f"      ... and {len(vit_models) - 5} more")

print("\n   b) DeiT models:")
deit_models = timm.list_models('deit*', pretrained=True)
print(f"      Found {len(deit_models)} DeiT models")
for model in deit_models:
    print(f"      - {model}")

print("\n   c) Swin Transformer:")
swin_models = timm.list_models('swin*', pretrained=True)
print(f"      Found {len(swin_models)} Swin models")
for i, model in enumerate(swin_models[:5]):
    print(f"      - {model}")
if len(swin_models) > 5:
    print(f"      ... and {len(swin_models) - 5} more")

print("\n   d) ResNet models:")
resnet_models = timm.list_models('resnet*', pretrained=True)
print(f"      Found {len(resnet_models)} ResNet models")
for i, model in enumerate(resnet_models[:5]):
    print(f"      - {model}")
if len(resnet_models) > 5:
    print(f"      ... and {len(resnet_models) - 5} more")

print("\n   e) EfficientNet models:")
effnet_models = timm.list_models('efficientnet*', pretrained=True)
print(f"      Found {len(effnet_models)} EfficientNet models")
for i, model in enumerate(effnet_models[:5]):
    print(f"      - {model}")
if len(effnet_models) > 5:
    print(f"      ... and {len(effnet_models) - 5} more")

print("\n   f) RegNet models:")
regnet_models = timm.list_models('regnet*', pretrained=True)
print(f"      Found {len(regnet_models)} RegNet models")
for i, model in enumerate(regnet_models[:8]):
    print(f"      - {model}")
if len(regnet_models) > 8:
    print(f"      ... and {len(regnet_models) - 8} more")


 Search for specific models:

   a) Vision Transformers (ViT):
      Found 476 ViT models
      - convit_base.fb_in1k
      - convit_small.fb_in1k
      - convit_tiny.fb_in1k
      - crossvit_9_240.in1k
      - crossvit_9_dagger_240.in1k
      ... and 471 more

   b) DeiT models:
      Found 24 DeiT models
      - deit3_base_patch16_224.fb_in1k
      - deit3_base_patch16_224.fb_in22k_ft_in1k
      - deit3_base_patch16_384.fb_in1k
      - deit3_base_patch16_384.fb_in22k_ft_in1k
      - deit3_huge_patch14_224.fb_in1k
      - deit3_huge_patch14_224.fb_in22k_ft_in1k
      - deit3_large_patch16_224.fb_in1k
      - deit3_large_patch16_224.fb_in22k_ft_in1k
      - deit3_large_patch16_384.fb_in1k
      - deit3_large_patch16_384.fb_in22k_ft_in1k
      - deit3_medium_patch16_224.fb_in1k
      - deit3_medium_patch16_224.fb_in22k_ft_in1k
      - deit3_small_patch16_224.fb_in1k
      - deit3_small_patch16_224.fb_in22k_ft_in1k
      - deit3_small_patch16_384.fb_in1k
      - deit3_small_patch16_384.

#### Search with multiple patterns

In [13]:
print("\n Search with filters:")
print("\n   Transformer models (various types):")
transformer_keywords = ['vit', 'deit', 'swin', 'beit', 'mae']
for keyword in transformer_keywords:
    models = timm.list_models(f'*{keyword}*', pretrained=True)
    if models:
        print(f"   - {keyword.upper()}: {len(models)} models")


 Search with filters:

   Transformer models (various types):
   - VIT: 476 models
   - DEIT: 25 models
   - SWIN: 38 models
   - BEIT: 13 models
   - MAE: 41 models


## Get Model Information

1. timm.list_models()                      # List all models
2. timm.list_models(pretrained=True)       # List pretrained models only
3. timm.list_models('resnet*')             # Wildcard search
4. timm.list_models('*vit*')               # Search containing 'vit'
5. timm.create_model('model_name')         # Create a model
6. timm.list_models(filter='keyword')      # Alternative filter syntax

Examples:

```python
  timm.list_models('vit_*_patch16_224')    # Specific pattern
  timm.list_models('*convnext*')           # ConvNeXt models
  timm.list_models('*mobile*')             # MobileNet variants
```

Use interactive Python to explore:

```python
import timm
models = timm.list_models('swin*', pretrained=True)
for m in models: 
    print(m)
```

## Get Detailed Model Info:
Example: Getting info for 'resnet50'

In [14]:
                    
try:
    model = timm.create_model('resnet50', pretrained=False)
    print(f"   - Architecture: ResNet-50")
    print(f"   - Input size: {model.default_cfg.get('input_size', 'N/A')}")
    print(f"   - Number of classes: {model.num_classes}")
    num_params = sum(p.numel() for p in model.parameters())
    print(f"   - Parameters: {num_params:,}")
except Exception as e:
    print(f"   Error: {e}")

   - Architecture: ResNet-50
   - Input size: (3, 224, 224)
   - Number of classes: 1000
   - Parameters: 25,557,032


## Model Naming Conventions Get Model Parameters

Example: 'regnetv_040.ra3_in1k'
Naming breakdown:

- `'regnetv'`    = Model architecture (RegNet variant V)
- `'40'`        = Model size/capacity (4.0 GFLOPs)
- `'ra3'`        = Training recipe/augmentation (RandAugment v3)
- `'in1k'`       = Dataset (ImageNet-1k)

Common naming patterns:

- Model sizes: `'tiny'`, `'small'`, `'base'`, `'large'`, or numbers like `'040'`, `'160'`
- Patch sizes: `'patch16'`, `'patch32'` (for transformers)
- Resolution: `'224'`, `'384'`, `'512'` (input image size)
- Datasets: `'in1k'` (ImageNet-1k), `'in21k'` (ImageNet-21k)
- Training: `'ra'` (RandAugment), `'dist'` (distilled), `'ft'` (fine-tuned)




In [15]:
# Example: regnetv_040.ra3_in1k
try:
    print("\nAnalyzing: regnetv_040.ra3_in1k")
    model = timm.create_model('regnetv_040.ra3_in1k', pretrained=False)
    
    # Count parameters
    total_params = sum(p.numel() for p in model.parameters())
    trainable_params = sum(p.numel() for p in model.parameters() if p.requires_grad)
    
    print(f"  ✓ Total parameters: {total_params:,} ({total_params/1e6:.2f}M)")
    print(f"  ✓ Trainable parameters: {trainable_params:,} ({trainable_params/1e6:.2f}M)")
    print(f"  ✓ Input size: {model.default_cfg.get('input_size', 'N/A')}")
    print(f"  ✓ Number of classes: {model.num_classes}")
    print(f"  ✓ Architecture: {model.__class__.__name__}")
    
except Exception as e:
    print(f"  ✗ Error: {e}")




Analyzing: regnetv_040.ra3_in1k
  ✓ Total parameters: 20,640,640 (20.64M)
  ✓ Trainable parameters: 20,640,640 (20.64M)
  ✓ Input size: (3, 224, 224)
  ✓ Number of classes: 1000
  ✓ Architecture: RegNet


## Compare Different Model Sizes

In [16]:
regnet_examples = ['regnetv_040.ra3_in1k', 'regnetv_064.ra3_in1k', 'regnetv_080.ra3_in1k']
for model_name in regnet_examples:
    try:
        model = timm.create_model(model_name, pretrained=False)
        params = sum(p.numel() for p in model.parameters())
        print(f"  {model_name:30s} → {params/1e6:6.2f}M parameters")
    except:
        print(f"  {model_name:30s} → Not available")

  regnetv_040.ra3_in1k           →  20.64M parameters
  regnetv_064.ra3_in1k           →  30.58M parameters
  regnetv_080.ra3_in1k           → Not available


## How to get model info for any model:

```python
import timm

# Create model (pretrained=False is faster for just checking params)
model = timm.create_model('model_name', pretrained=False)

# Count parameters
total_params = sum(p.numel() for p in model.parameters())
print(f"Parameters: {total_params:,} ({total_params/1e6:.2f}M)")

# Get configuration
print(f"Input size: {model.default_cfg['input_size']}")
print(f"Classes: {model.num_classes}")
""")
```